src/store/ChartStore.js

```diff
@@ -307,6 +311,15 @@ class ChartStore {
         this.stxx = null;
     }
 
+    @action.bound onReceiveProps(nextProps) {
+        if (!this.context) return;
+
+        const { symbol } = nextProps;
+        if (symbol && this.activeSymbols.length > 0 && this.currentActiveSymbol.symbol !== symbol) {
```

Comment:
When the context is not yet ready or the activeSymbols are empty, can we keep symbol value and use it to render the chart?

```javascript
+        api.getActiveSymbols().then(({ active_symbols }) => {
+            this.setActiveSymbols(active_symbols);
+            if (symbol) {
+                this.changeSymbol(symbol);
+            } else if (stxx.chart.symbol) {
+                this.currentActiveSymbol = stxx.chart.symbolObject;
+            } else {
+                this.changeSymbol(defaultSymbol); // use is here instead of defaultSymbol
+            }
+        });
```

Reply:
This is not possible because when we set the current symbol, we need the name, market, submarket names, and pip size as well. What this means is that the chart will render, but the chart title will be blank.

Unless... instead of having the user just pass the symbol code, we have them pass the symbol object itself.